pkg/judge/judge.go

```diff
@@ -0,0 +1,46 @@
+package judge
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/jasondellaluce/synchro/pkg/utils"
+)
+
+func Judge(ctx context.Context, git utils.GitHelper, commit string) error {
+	out, err := git.DoOutput("show", "--raw", commit)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(out, "\n")
+
+	modified := 0
+	added := 0
+	deleted := 0
+	renamed := 0
+
+	for _, line := range lines {
+		//only check file metadata lines
+		if len(line) > 0 && line[0] == ':' {
+			//file mode is the 32nd character of the line
+			switch line[31] {
+			case 'M':
+				modified = 1
+			case 'A':
+				added = 1
+			case 'D':
+				deleted = 1
+			case 'R':
+				renamed = 1
+			}
+		}
+	}
+
+	if (modified + added + deleted + renamed) > 1 {
+		return fmt.Errorf("commits can either have modified, added, deleted or renamed files and not a combination of those")
```

Comment:
Like for other commands, I would pair the returned error with a `fmt.Fprintf(os.Stderr, "...")`, so that when checking the status code of the command one could simply grab the stdout to understand if the tool suggested what's wrong. This pattern is used across the other commands. In this case, I would write a more detailed message in the output, stating exactly what types of modifications are overlapping in the commit patch. It would be also cool to collect the file names for each, static what has been modified and what added, etc...